Add trustworthiness score for dimred task

michalk8 opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Submission type

* [ ] This submission adds a new dataset
* [ ] This submission adds a new method
* [x] This submission adds a new metric
* [ ] This submission adds a new task
* [ ] This submission adds a new Docker image
* [ ] This submission fixes a bug (link to related issue: )
* [ ] This submission adds a new feature not listed above

### Testing

* [x] This submission was written on a forked copy of SingleCellOpenProblems
* [x] GitHub Actions "Run Benchmark" tests are passing on this base branch of this pull request (include link to passed test: )
  https://github.com/michalk8/SingleCellOpenProblems/actions/runs/698892182
* [x] If this pull request is not ready for review (including passing the "Run Benchmark" tests), I will open this PR as a draft (click on the down arrow next to the "Create Pull Request" button)

### Submission guidelines

* [x] This submission follows the guidelines in our [Contributing](../blob/master/CONTRIBUTING.md) document
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change

This PR adds trustworthiness score from `sklearn`.
Things that need to be resolved:
 - the method does not work if there are non-finite values in the embedding/original space. Current solution: if original space contains non-fin. values (e.g. NaN), return NaN,
if the embedding contains these values, 0 is return (lowest possible value for the score).
- locally, I've tested that it works also on sparse matrices (in case adata.X), but haven't added any test for this: should the test be added in `test_metrics.py`?
- following the RMSE-based score, I've also saved the trust. score to `adata.uns['trustworthiness_score']` - is this desired?

closes #254

michalk8 commented:
pinging @dburkhardt w.r.t. to the above mentioned things to be resolved.

LuckyMD reviewed (approved):
looks good to me. If scaling becomes an issue we can think about disabling i guess...

LuckyMD commented:
As for your points:
* at the moment we don't really monitor the intermediate stored objects. This might change in the future, but scores don't really need to be saved in the object.
* I would add a test for sparse matrices
* Non-finite values in the high dimensional space should be removed in the data loader. I think that's not sth this metric should be worrying about.

michalk8 commented:
Thanks for the feedback, I've addressed all of the 3 points:
- not storing the score in `adata` anymore
- I've added 1 test to `test_metrics.py` for sparse `adata.X`
- remove the check for non-finite values in `adata.X`

LuckyMD commented:
What do you think @dburkhardt @scottgigante? Maybe easier to add a sparse test dataset?

michalk8 commented:
> i would re-add the comment about metric faults. That will help future people look at the code.

I'd say it's self-explanatory, esp. after removing checks for non-finite in `adata.X`.

michalk8 commented:
Before I forget, I suggest squashing the commits from PRs (easiest is to change the merge button in a PR to squash merge).

dburkhardt reviewed (commented):
This all looks good. The last thing that needs to happen is the README.md needs to be updated to describe the newly added metric. I would be fine with copying from https://scikit-learn.org/stable/modules/generated/sklearn.manifold.trustworthiness.html.
